lib/fs.js

```diff
@@ -2479,3 +2479,457 @@ Object.defineProperty(fs, 'SyncWriteStream', {
   set: internalUtil.deprecate((val) => { SyncWriteStream = val; },
                               'fs.SyncWriteStream is deprecated.', 'DEP0061')
 });
+
+
+// The Promises API
+
+async function* readEverything(fd) {
+  let eof = false;
+  while (!eof) {
+    const buffer = Buffer.allocUnsafe(kReadFileBufferLength);
+    const { length } =
+      await fs.promises.read(fd, buffer, 0, buffer.length);
+    if (length < buffer.length)
+      eof = true;
+    yield buffer.slice(0, length);
+  }
+}
```

Comment:
This is going to be very slow. Do we know if there is some improvements in the area from the V8 team? 
Can you use the same class approach I used for Async Iterators? It ended up being 2x the async generator function.

Reply:
Yes this can use the class approach, but I'm purposefully leaving this in so that folks my @bmeurer can take a look and figure out how to make it faster. I will be switching it to the class approach a bit later.